inference-engine/src/cldnn_engine/cldnn_engine.cpp

```diff
@@ -270,72 +271,207 @@ void clDNNEngine::SetConfig(const std::map<std::string, std::string> &config) {
     _impl->m_config.UpdateFromMap(config);
 }
 
-void clDNNEngine::QueryNetwork(const ICNNNetwork& network, const std::map<std::string, std::string>& config, QueryNetworkResult& res) const {
-    std::vector <CNNLayer::Ptr> concats;
-    std::vector <CNNLayer::Ptr> nextLayerDependent;
-
-    // Verify device id
-    GetDeviceInfo(config);
-
-    if (network.getFunction()) {
-        THROW_IE_EXCEPTION << NOT_IMPLEMENTED_str << " ngraph::Function is not supported natively";
-    }
+void clDNNEngine::QueryNetwork(const ICNNNetwork& network,
+                               const std::map<std::string,
+                               std::string>& config,
+                               QueryNetworkResult& res) const {
+    GetDeviceInfo(config);      // Verify device id
+    auto function = network.getFunction();
+    if (function != nullptr) {
+        std::unordered_set<std::string> originalOps;
+        for (auto&& node : function->get_ops()) {
+            if (!ngraph::op::is_parameter(node) &&
+                !ngraph::op::is_output(node)) {
+                originalOps.emplace(node->get_friendly_name());
+            }
+        }
+        auto clonedNetwork = CloneNetwork(network);
+        std::unordered_set<std::string> supported;
+        std::unordered_set<std::string> unsupported;
+
+        std::unordered_set<std::string> splitNames;
+        std::unordered_set<std::string> concatNames;
+        std::unordered_set<std::string> depLayerNames;
+
+        std::vector<std::shared_ptr<ngraph::Node>> splits;
+        std::vector<std::shared_ptr<ngraph::Node>> concats;
+        std::vector<std::shared_ptr<ngraph::Node>> nextLayerDependent;
+        
+        for (CNNNetworkIterator itLayer{clonedNetwork.get()};
```

Comment:
does it really work? No conversion from ngraph to old representation is applied

Reply:
Sorry, I didn't catch what you mean exactly.
If you mean CNNLayer objects creation, than it was done by executing convertFunctionToICNNNetwork() inside CloneNetwork() method, which was called above.
The code definitely works - added tests pass and any v10 IR I've tested so far passed successfully.

Reply:
> executing convertFunctionToICNNNetwork() inside CloneNetwork() method

This is not right, `CloneNetwork` does not perform conversion. Implicit conversion is removed from IE (except `setBatchSize`). See https://github.com/openvinotoolkit/openvino/blob/master/inference-engine/src/legacy_api/src/ie_util_internal.cpp#L153-L161, not conversion is applied

Reply:
Done.